command/node_drain_test.go

```diff
@@ -237,6 +237,19 @@ func TestNodeDrainCommand_Monitor(t *testing.T) {
 	if !strings.HasSuffix(out, expected) {
 		t.Fatalf("expected output to end with:\n%s", expected)
 	}
+
+	// Test -monitor flag
+	outBuf.Reset()
+	args = []string{"-address=" + url, "-self", "-monitor", "-ignore-system"}
+	t.Logf("Running: %v", args)
+	if code := cmd.Run(args); code != 0 {
+		t.Fatalf("expected exit 0, got: %d\n%s", code, outBuf.String())
+	}
+
+	out = outBuf.String()
+	t.Logf("Output:\n%s", out)
+
+	require.Contains(out, "marked all allocations for migration")
```

Comment:
Hmm that is pretty jank. If you run `nomad node drain -monitor <node that has never been marked for drain>` it will say that all allocations have been marked for migration even if that isn't true.

I think we need to modify this (https://github.com/hashicorp/nomad/blob/51440e24a7886986d74fcb94320707842adcb39c/api/nodes.go) to track whether the DrainStrategy was ever non-nil and then if it becomes nil it can print that message. Otherwise if it is nil and has never been set to a non-nil strategy we can print a more generic message like, "no drain strategy set"

Reply:
fixed in a66d89b